mindmeld/components/question_answerer.py

```diff
@@ -449,805 +591,2172 @@ def _generate_mapping_data(embedder_model, embedding_fields):
-                origin = self.location
-                scale = "5km"
-            else:
-                max_value = self.field_stats["max_value"]
-                min_value = self.field_stats["min_value"]
-
-                if self.field_info.is_date_field():
-                    # ensure the timestamps for date fields are integer values
-                    max_value = int(max_value)
-                    min_value = int(min_value)
-
-                    # add time unit for date field
-                    scale = (
-                        "{}ms".format(int(0.5 * (max_value - min_value)))
-                        if max_value != min_value
-                        else 1
+            elif self.data_type in ["bool"]:
+                if not isinstance(boolean, bool):
+                    raise ValueError(
+                        "Invalid boolean parameter."
                     )
-                else:
-                    scale = (
-                        0.5 * (max_value - min_value) if max_value != min_value else 1
+            elif self.data_type in ["string"]:
+                if not self.is_string(filter_text):
+                    raise ValueError(
+                        "Invalid textual input parameter."
                     )
+            else:
+                raise ValueError(
+                    "Custom filter can only be defined for boolean, number, string or date field."
+                )
 
-                if self.sort_type == "asc":
-                    origin = min_value
-                else:
-                    origin = max_value
+        def do_filter(self, curated_docs,
+                      filter_text=None,
+                      gt=None, gte=None, lt=None, lte=None, et=None, boolean=None):
+            """
+            Filters a list of docs to a subset based on some criteria such as a boolean value
+            or ><= operations or a text snippet.
+            """
 
-            sort_clause = {
-                "linear": {self.field: {"origin": origin, "scale": scale}},
-                "weight": self.DEFAULT_SORT_WEIGHT,
-            }
+            if not curated_docs:
+                return curated_docs
+
+            self.do_filter_validation(filter_text, gt, gte, lt, lte, et, boolean)
+
+            if self.data_type in ["number"]:
+
+                def is_valid(value):
+                    if gt and not (value > gt):
+                        return False
+                    if gte and not (value >= gte):
+                        return False
+                    if lt and not (value < lt):
+                        return False
+                    if lte and not (value <= lte):
+                        return False
+                    if et and not (value != et):
+                        return False
+                    return True
+
+            elif self.data_type in ["date"]:
+
+                def is_valid(value):
+                    value = abs(self.date_scorer(value))  # returns no. of days
```

Comment:
seems like date and number have similar logic that can be collapsed. Also, could you explain what the conditional logic is doing?